tiledb/sm/query/dimension_label/array_dimension_label_queries.h

```diff
@@ -134,36 +143,40 @@ class ArrayDimensionLabelQueries {
   /** The storage manager. */
   StorageManager* storage_manager_;
 
-  /** TODO: Docs */
+  /** Class stats object for timing. */
   stats::Stats* stats_;
 
   /** Map from label name to dimension label opened by this query. */
   std::unordered_map<std::string, tdb_unique_ptr<DimensionLabel>>
       dimension_labels_;
 
-  /** Map from label name to dimension label range query. */
-  std::unordered_map<std::string, tdb_unique_ptr<DimensionLabelQuery>>
-      range_queries_map_;
+  /** Dimension label range queries */
+  std::vector<tdb_unique_ptr<DimensionLabelRangeQuery>> range_queries_;
 
   /**
-   * Non-owning vector for accessing query by dimension index.
+   * Non-owning vector for accessing range query by dimension index.
    *
    * Note: This vector is always sized to the number of dimensions in the array.
    * There can be at most on query per dimension. If there is no query on the
    * dimension, it contains a nullpointer.
    */
-  std::vector<DimensionLabelQuery*> range_queries_;
+  std::vector<DimensionLabelRangeQuery*> range_queries_by_dim_idx_;
 
   /**
    * Dimension label data queries.
    *
    * Note: For the data queries, the element order to the queries is
    * unimportant and does not correspond to dimension index or any other value.
    */
-  std::vector<tdb_unique_ptr<DimensionLabelQuery>> data_queries_;
+  std::vector<tdb_unique_ptr<DimensionLabelDataQuery>> data_queries_;
 
-  /** Non-owning map from label name to dimension label data query. */
-  std::unordered_map<std::string, DimensionLabelQuery*> data_queries_map_;
+  /**
+   * Non-owning vector for accessing dat query by dimension index.
+   *
+   * Note: The outer vector is always sized to the number of dimensions in the
+   * array. The internal vector will grow as queries are added.
+   */
+  std::vector<std::vector<DimensionLabelDataQuery*>> data_queries_by_dim_idx_;
```

Comment:
This is for label data correct? Could we rename to label_data_queries_by_dim_idx_?

Reply:
Can we add a similar note for nullptr like for range_queries_by_dim_idx_?

Reply:
I'll rename them.

This one doesn't have a nullptr placeholder. There can be multiple dimension label data queries on a single dimension, so it's a vector of vectors instead of a vector with nullptr as a placeholder. I will add additional comments to bring attention to this distinction.